autorest/responder_test.go

```diff
@@ -423,6 +423,45 @@ func TestRespondAcceptsNullResponse(t *testing.T) {
 	}
 }
 
+func TestWithServiceErrorUnlessOK_OKResponse(t *testing.T) {
+	v := &mocks.T{}
+	var e interface{}
+	r := mocks.NewResponseWithContent(jsonT)
+	err := Respond(r,
+		WithServiceErrorUnlessOK(v, &e),
+		ByClosing())
+	if err != nil {
+		t.Fatalf("autorest: WithServiceErrorUnlessOK failed (%v)", err)
+	}
+	if v.Name != "Rob Pike" || v.Age != 42 {
+		t.Fatalf("autorest: WithServiceErrorUnlessOK failed to properly unmarshal")
+	}
+}
+
+func TestWithServiceErrorUnlessOK_ErrorResponse(t *testing.T) {
+	v := &mocks.T{}
+	e := &mocks.T{}
+	r := mocks.NewResponseWithContent(jsonT)
+	r.Status = "400 BadRequest"
+	r.StatusCode = http.StatusBadRequest
+
+	err := Respond(r,
+		WithServiceErrorUnlessOK(v, e),
+		ByClosing())
+	if err == nil {
+		t.Fatalf("autorest: WithServiceErrorUnlessOK failed to return error on bad request")
+	}
+
+	if e.Name != "Rob Pike" || e.Age != 42 {
```

Comment:
this test is not easy to follow. please do not share `mocks.T` type for both "value" and "error" types. You're not testing if things still get deserialized into `v`.